app/javascript/controllers/ui/filter_controller.js

```diff
@@ -2,18 +2,21 @@ import { Controller } from "@hotwired/stimulus";
 
 export default class UIFilter extends Controller {
   static targets = ["source", "item"];
+  static values = {
+    pattern: String,
+  };
 
   connect() {}
 
   filter(event) {
     let lowerCaseFilterTerm = this.sourceTarget.value.toLowerCase();
-    let regex = new RegExp("^" + lowerCaseFilterTerm);
+    const pattern = this.patternValue.replace("{input}", lowerCaseFilterTerm);
+    const regex = new RegExp(pattern);
```

Comment:
Except for readability, I think the pattern variable is not necessary.
```suggestion
    const regex = new RegExp(this.patternValue.replace("{input}", lowerCaseFilterTerm));
```

Reply:
Just a suggestion, if you think it is necessary keep it.

Reply:
nahh, it was just for readability.